src/components/Checkbox.tsx

```diff
@@ -0,0 +1,123 @@
+import type { CheckboxProps, ToggleProps } from '@react-types/checkbox';
+import { gsap } from 'gsap';
+import type { ComponentPropsWithRef } from 'react';
+import { useEffect, useRef } from 'react';
+import { useCheckbox } from 'react-aria';
+import { useToggleState } from 'react-stately';
+import { styled } from '../../stitches.config';
+import { CheckIcon } from '../animationIcons/CheckIcon';
+
+const Label = styled('label', {
+  '& input:checked:disabled + div': {
+    color: '$gray50',
+  },
+  display: 'block',
+  position: 'relative',
+});
+
+const Input = styled('input', {
+  '&:checked': {
+    backgroundColor: '$brandYellow',
+    borderColor: '$brandYellow',
+  },
+  '&:checked&:disabled': {
+    backgroundColor: '$yellow20',
+    borderColor: '$yellow20',
+  },
+  '&:disabled': {
+    borderColor: '$gray30',
+  },
+  '&:focus': {
+    outline: '2px solid $brandYellow',
+    outlineOffset: '2px',
+  },
+  '&:indeterminate': {
+    borderColor: '$brandYellow',
+    position: 'relative',
+  },
+  '&:indeterminate:after': {
+    backgroundColor: '$brandYellow',
+    content: '""',
+    height: '12px',
+    left: '50%',
+    position: 'absolute',
+    top: '50%',
+    transform: 'translate(-50%, -50%)',
+    width: '12px',
+  },
+  '&:invalid': {
+    borderColor: '$uiErrorRegular',
+  },
+  appearance: 'none',
+  backgroundColor: '$brandWhite',
+  borderColor: '$gray50',
+  borderRadius: '$4',
+  borderStyle: '$solid',
+  borderWidth: '$thin',
+  height: '24px',
+  margin: '0',
+  position: 'relative',
+  width: '24px',
+});
+
+const IconContainer = styled('div', {
+  left: '50%',
+  position: 'absolute',
+  top: '50%',
+  transform: 'translate(-50%, -50%)',
+});
+
+type CustomCheckboxProps = {
+  text?: string;
+};
+
+export const Checkbox = ({
+  validationState,
+  text,
+  defaultSelected,
+  isIndeterminate,
+  ...restProps
```

Comment:
I usually highly advise not using rest props on component libraries like this, and to specifically list all props out so you can make sure you are never spreading props improperly to base react components (if you did you'll get those annoying react warnings about a prop not being recognized).

In general it would be very easy for a consumer to pass a prop that _could_ ultimately end up getting inappropriately passed all the way down to base component that it shouldnt.